### PR TITLE
add codebuild ymls and makefile DLTP-1717, ESU-1132

### PR DESCRIPTION
**Pull request by msuhovec**
I'm adding the codebuild yaml. The AWS and github hooks are not there yet.

### CODE REVIEW

**Comment by dbrower** [comment]
Does this code build config use a docker image to do the build?

File: codepipeline.yml [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+    Type: String
+    Default: curatend-batch
+
+  GitHubRepoBranch:
+    Type: String
+    Default: master
+
+  CodeBuildComputeType:
+    Type: String
+    Default: BUILD_GENERAL1_MEDIUM
```
Comment (ialford) [inline codepipeline.yml]: I'm wondering if we need a Medium here - we could check performance before committing to this size. I believe it was the most cost-effective at the time for Bendo, but we should revisit.

File: codepipeline.yml [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+              Service: "codebuild.amazonaws.com"
+        Version: "2012-10-17"
+
+  BuildProjectServicePolicy:
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyName: "CodeBuildServicePolicy"
+      PolicyDocument:
+        Statement: 
+          - Action: "s3:*"
```
Comment (ialford) [inline codepipeline.yml]: Does the CodeBuild need to delete objects? We should probably pare these permissions down.

File: codepipeline.yml [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+            Effect: "Allow"
+            Principal: 
+              Service: "codepipeline.amazonaws.com"
+        Version: "2012-10-17"
+      Policies:
+        - PolicyName: "ReadWriteArtifacts"
+          PolicyDocument:
+            Version: "2012-10-17"
+            Statement:
+              Action: "s3:*"
```
Comment (ialford) [inline codepipeline.yml]: See the CodeBuild comment, and possibly scale back these permissions.

File: codepipeline.yml (hunk 2 above)
Comment (jgondron) [reply codepipeline.yml]: I can, but if you follow cdk's lead, it's giving a lot of permissions to the artifact bucket: https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-codepipeline-actions/test/integ.pipeline-code-build-multiple-inputs-outputs.expected.json#L554. 🤷‍♂

**Pull request merged**